OpenSim/Simulation/Model/Model.cpp

```diff
@@ -1916,10 +1926,9 @@ const Vector& Model::getControls(const SimTK::State &s) const
 /** Compute the controls the model */
 void Model::computeControls(const SimTK::State& s, SimTK::Vector &controls) const
 {
-    for (auto& controller : getComponentList<Controller>()) {
-        if (controller.isEnabled()) {
-            controller.computeControls(s, controls);
-        }
+    for (const auto& controllerRef : this->_enabledControllers) {
```

Comment:
your can guard this `for` with:
```cpp
if( getAllControllersEnabled() ){ 
...
}
```
which then allows the Model to override and disable all controllers (essentially skip calling computeControls() on individual controllers). Sorry that Controller logic needed to be fixed ages ago.